### PR TITLE
Avoid warnings in bundled muparser and boost

### PR DESCRIPTION
**Pull request by masterleinad**
This should remove the last warnings we get using gcc-7.1.0 with bundled packages.

### CODE REVIEW

**Comment by drwells** [comment]
Do we need to add our own default copy constructor?

**Comment by tamiko** [comment]
@drwells No, the default copy constructor should be fine, shouldn't it?

File: bundled/boost-1.62.0/include/boost/signals2/detail/auto_buffer.hpp [hunk 1]
```diff
@@ -304,6 +304,7 @@ namespace detail
             (*this).~auto_buffer();
             buffer_   = new_buffer;
             members_.capacity_ = new_capacity;
+            size_ = std::max(static_cast<decltype(size_)>(0),size_);
```
Comment (bangerth) [inline bundled/boost-1.62.0/include/boost/signals2/detail/auto_buffer.hpp]: Can you explain why this change is correct? I mean, I have no idea of what the code does, it just looks funny to add a line in a random place in boost :-)

**Comment by bangerth** [comment]
Can you copy what the warning is that this patch is trying to address?

**Comment by drwells** [comment]
@tamiko Yes, you are right. Why does the current copy constructor set `m_end_of_sequence` unconditionally to `false`? Will we break something if we copy a `true`?

**Comment by tamiko** [comment]
@drwells I do not know. :-)

**Comment by tamiko** [comment]
I think this PR is good to merge. @drwells @bangerth ?

**Comment by masterleinad** [comment]
@bangerth Are you fine with the last change?

**Comment by bangerth** [comment]
Yes, let's do it!

Do you want to open the issue for boost, or do you want me to?

**Comment by masterleinad** [comment]
In the current [developer version](https://github.com/boostorg/signals2/blob/develop/include/boost/signals2/detail/auto_buffer.hpp#L308) the destructor is no longer called, but a member function is used instead. So it seems that there is no need for an issue in the end.

**Comment by bangerth** [comment]
Ah, nice. Thanks for double checking!
